### DISCUSSION TITLE
Switch toggle in table rows?

### DISCUSSION THREAD

**Discussion by mtl87** [twilio-labs/paste#3054] [Q&A] [answered]
Can a switch live in a table row or only checkboxes? See attached photos. Thank you!

![table-checkbox](https://user-images.githubusercontent.com/8579607/221259975-7679efe1-0996-49a9-b9aa-e95d4d761e8f.png)

![table-switch](https://user-images.githubusercontent.com/8579607/221259980-334826a6-97c4-43be-a070-7de82b169307.png)

**Comment by nkrantz** [comment, marked as answer]
Hey, @mtl87! Tables should only contain static data ([see Table docs here](https://paste.twilio.design/components/table#accessibility)) because interactions on large Tables can cause unpleasant experiences for some assistive technology/keyboard users. If you need to put interactive controls in your Table, we'd suggest using a [Data Grid](https://paste.twilio.design/components/data-grid#about-data-grid) instead. The Data Grid only has one tab stop, so users don't have to tab through every single control in order to escape the table.

Switches are fine to use in Data Grids, as are Checkboxes. If you do use the Switch, just be sure that the label text ("Enable" in your screenshot) doesn't change based on the state of the Switch. For example, when a user clicks a Switch, that label text shouldn't change to "Disable" ([read more on Switch docs](https://paste.twilio.design/components/switch#about-switch)).

**Reply by nkrantz** [reply]
A bit more guidance here. We missed the "Save geo permissions" Button in your mockups the first time around. Since you have that button to save any changes, I'd recommend definitely going with the Checkboxes instead of Switches here. Also, since the page is titled "Messaging Geographic Permissions", you could label the Data Grid column header "Enable permissions" instead of "Geo permissions" as it might be a bit redundant. Let us know if you have any more questions!

**Reply by mtl87** [reply]
Thank you for the follow up and insight :)
